javascript/javascript1/week2/student-manager.js

```diff
@@ -0,0 +1,61 @@
+
+const class07Students = [];
+function addStudentToClass(studentName) {
+    if (!studentName) {
+        console.log(`Please write down your name`)
+        return
+    }
+
+    if (class07Students.length < 6 || studentName === 'Queen') {
+        
+        if (class07Students.includes(studentName)) {
+            console.log(`Student ${studentName} is already in the class`)
+        }else {
+            return class07Students.push(studentName)
+        } 
+    }else {
+        console.log('Cannot add more students to class 07') 
+        }
```

Comment:
I see I confused you with giving you this to start on `class07Students.length < 6 || studentName === 'Queen'` (which was wrong by me).

I'm going to show you what I mean:
```js
    if (!studentName) {
        console.log(`"Please write down your name"`)
        return
    }

    if (class07Students.length >= 6 && studentName !== 'Queen') {
        console.log('Cannot add more students to class 07')
        return
    }

    if (class07Students.includes(studentName)) {
        console.log(`Student ${studentName} is already in the class`)
        return
    }

    class07Students.push(studentName)
}
```

Notice single nesting and how you are able to do one check at the time and return.

Also notice how rules can be "inverted" (`class07Students.length < 6 || studentName === 'Queen'`), see above :point_up: .

Reply:
thank you for comments 💯 :). I learned so much from them.

Reply:
You are welcome